### PR TITLE
Reformat references

### PR DESCRIPTION
**Pull request by timhoffm**
Follow-up to #19774.

### CODE REVIEW

File: examples/images_contours_and_fields/figimage_demo.py [hunk 1]
```diff
@@ -22,14 +21,11 @@
...
 #
-# The use of the following functions, methods, classes and modules is shown
-# in this example:
-
-matplotlib.figure.Figure
-matplotlib.figure.Figure.figimage
-matplotlib.pyplot.figimage
+#    - `matplotlib.figure.Figure`
+#    - `matplotlib.figure.Figure.figimage`
+#    - `matplotlib.pyplot.figimage`
```
Comment (QuLogic) [inline examples/images_contours_and_fields/figimage_demo.py:31]: Can join these?

File: examples/lines_bars_and_markers/filled_step.py [hunk 2]
```diff
@@ -225,15 +225,11 @@ def stack_hist(ax, stacked_data, sty_cycle, bottoms=None,
...
 #
-# The use of the following functions, methods, classes and modules is shown
-# in this example:
-
-import matplotlib
-matplotlib.axes.Axes.fill_betweenx
-matplotlib.axes.Axes.fill_between
-matplotlib.axis.Axis.set_major_locator
+#    - `matplotlib.axes.Axes.fill_betweenx` / `matplotlib.pyplot.fill_betweenx`
+#    - `matplotlib.axes.Axes.fill_between` / `matplotlib.pyplot.fill_betweenx`
```
Comment (QuLogic) [inline examples/lines_bars_and_markers/filled_step.py:234]:
```suggestion
#    - `matplotlib.axes.Axes.fill_between` / `matplotlib.pyplot.fill_between`
```

File: examples/scales/scales.py [hunk 3]
```diff
@@ -103,20 +103,16 @@ def inverse(a):
...
-matplotlib.axes.Axes.set_yscale
-matplotlib.axes.Axes.set_xscale
-matplotlib.axis.Axis.set_major_locator
-matplotlib.scale.LogitScale
-matplotlib.scale.LogScale
-matplotlib.scale.LinearScale
-matplotlib.scale.SymmetricalLogScale
-matplotlib.scale.FuncScale
+#    - `matplotlib.axes.Axes.set_yscale`
+#    - `matplotlib.axes.Axes.set_xscale`
```
Comment (QuLogic) [inline examples/scales/scales.py:112]:
```suggestion
#    - `matplotlib.axes.Axes.set_xscale`
#    - `matplotlib.axes.Axes.set_yscale`
```

File: examples/scales/scales.py [hunk 4]
```diff
@@ -103,20 +103,16 @@ def inverse(a):
...
+#    - `matplotlib.axis.Axis.set_major_locator`
+#    - `matplotlib.scale.LogitScale`
+#    - `matplotlib.scale.LogScale`
+#    - `matplotlib.scale.LinearScale`
+#    - `matplotlib.scale.SymmetricalLogScale`
+#    - `matplotlib.scale.FuncScale`
```
Comment (QuLogic) [inline examples/scales/scales.py:118]: Probably sort these too, unless this file is specifically in this order?

File: examples/shapes_and_collections/fancybox_demo.py [hunk 5]
```diff
@@ -101,19 +101,15 @@ def draw_control_points_for_patches(ax):
...
-matplotlib.patches.FancyBboxPatch
-matplotlib.patches.BoxStyle
-matplotlib.patches.BoxStyle.get_styles
-matplotlib.transforms.Bbox
-matplotlib.figure.Figure.text
-matplotlib.axes.Axes.text
+#    - `matplotlib.patches`
+#    - `matplotlib.patches.FancyBboxPatch`
+#    - `matplotlib.patches.BoxStyle`
+#    - ``matplotlib.patches.BoxStyle.get_style``
```
Comment (QuLogic) [inline examples/shapes_and_collections/fancybox_demo.py:112]: Should this be?
```suggestion
#    - `matplotlib.patches.BoxStyle.get_styles`
```

File: examples/shapes_and_collections/hatch_demo.py [hunk 6]
```diff
@@ -47,19 +47,15 @@
...
-matplotlib.patches.Patch.set_hatch
-matplotlib.axes.Axes.bar
-matplotlib.pyplot.bar
+#    - `matplotlib.patches`
+#    - `matplotlib.patches.Ellipse`
+#    - `matplotlib.patches.Polygon`
+#    - `matplotlib.axes.Axes.add_patch`
+#    - `matplotlib.patches.Patch.set_hatch`
+#    - `matplotlib.axes.Axes.bar`
+#    - `matplotlib.pyplot.bar`
```
Comment (QuLogic) [inline examples/shapes_and_collections/hatch_demo.py:61]: Join lines.

File: examples/text_labels_and_annotations/fancyarrow_demo.py [hunk 7]
```diff
@@ -57,16 +57,12 @@ def to_texstring(s):
...
-# in this example:
-
-import matplotlib
-matplotlib.patches
-matplotlib.patches.ArrowStyle
-matplotlib.patches.ArrowStyle.get_styles
-matplotlib.axes.Axes.annotate
+#    - `matplotlib.patches`
+#    - `matplotlib.patches.ArrowStyle`
+#    - ``matplotlib.patches.ArrowStyle.get_style``
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/fancyarrow_demo.py:67]: There's no `get_style` for this one either
```suggestion
#    - `matplotlib.patches.ArrowStyle.get_styles`
```

File: doc/sphinxext/redirect_from.py [hunk 8]
```diff
@@ -49,6 +49,7 @@ def setup(app):
     RedirectFrom.app = app
     app.add_directive("redirect-from", RedirectFrom)
     app.connect("build-finished", _generate_redirects)
+    return {'parallel_read_safe': True}
```
Comment (QuLogic) [inline doc/sphinxext/redirect_from.py:52]: Why is this change here?

Comment (timhoffm) [reply doc/sphinxext/redirect_from.py]: Sorry, this is my personal temporary workaround to have parallel local doc-build. (The correct way to get parallel doc builds back is #19686, but that needs still in a bit of work).

**Review by QuLogic — APPROVED** [review]

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Oh, hmmm, did we really want this back ported?
